libvast/vast/system/source.hpp

```diff
@@ -216,21 +221,30 @@ source(caf::stateful_actor<source_state<Reader>>* self, Reader reader,
        size_t table_slice_size, caf::optional<size_t> max_events,
        type_registry_type type_registry, vast::schema local_schema,
        std::string type_filter, accountant_type accountant) {
+  VAST_TRACE(VAST_ARG(self));
   // Initialize state.
   auto& st = self->state;
   st.init(self, std::move(reader), std::move(max_events),
           std::move(type_registry), std::move(local_schema),
           std::move(type_filter), std::move(accountant));
+  self->set_down_handler([=](const caf::down_msg& msg) {
+    VAST_VERBOSE(self, "received DOWN from", msg.source);
+    self->state.done = true;
+  });
```

Comment:
Why do we need this handler? The source doesn't monitor any actor, so it cannot receive a DOWN message.

Reply:
Leftover from testing where I monitored the node. Let's clean this up once we figured out the remaining shutdown issue.

Reply:
Fair enough; let's leave this comment open until you remove the handler.

Reply:
I removed it.